cmd/common/image_functions.go

```diff
@@ -196,6 +196,10 @@ func GetUntaggedManifests(ctx context.Context, acrClient api.AcrCLIClientInterfa
 		if err != nil {
 			return nil, err
 		}
+		if UntagLimit > 0 && len(candidates) >= UntagLimit {
+			// If the number of candidates is greater than the limit, we stop fetching more manifests.
+			break
```

Comment:
Breaking here means only a subset of manifest is scanned. It may cause unwanted purge since any currently dangling manifest might get referenced by a tag in the next iteration.